File: AlveoManagementServer/AlveoManagementCommon/Classes/Project_Classes/Task.cs
﻿using AlveoManagementCommon.Interfaces;

namespace AlveoManagementCommon.Classes
{
    public class Task : ITask
    {
        public string ID { get; set; }
        public string Name { get; set; }
        public string StartDate { get; set; }
        public string EndDate { get; set; }
        public string Number { get; set; }
        public int Progress { get; set; }
        public int Duration { get; set; }
        public string Parent { get; set; }
        public string Color { get; set; }
        public string Type { get; set; }
        public string Personnel { get; set; }
        public string Leader { get; set; }
    }
}

File: AlveoEnergy/src/app/models/task.ts
import { Guid } from 'src/app/classes/guid';


export class Tasks {
    public ID: Guid;
    public StartDate: string;
    public EndDate: string;
    public Name: string;
    public Progress: number;
    public Duration: number;
    public Parent: string;
    public Color: string;
    public Type: String;
    public Personnel: string;
    public Leader: string;
    public Number: string;

    constructor() {
    }

    ToJson(): string {
        return JSON.stringify(this);
    }

    static FromJson(jObj: any): Tasks {
        var project = new Tasks();
        project.ID = jObj.ID;
        project.StartDate = jObj.StartDate;
        project.EndDate = jObj.EndDate;
        project.Name = jObj.Name;
        project.Progress = jObj.Progress;
        project.Duration = jObj.Duration;
        project.Parent = jObj.Parent;
        project.Color = jObj.Color;
        project.Type = jObj.Type;
        project.Personnel = jObj.ProPersonnelgress;
        project.Leader = jObj.Leader;
        project.Number = jObj.Number;
        return project;
    }

}